### PR TITLE
CI: Allow macos builds to fail without canceling other C++ builds.

### PR DESCRIPTION
**Pull request by arthurp**

### CODE REVIEW

File: .github/workflows/cpp.yaml [hunk 1]
```diff
@@ -70,7 +70,9 @@ jobs:
   ###################################################################
   build_and_test:
     runs-on: ${{matrix.os}}
+    continue-on-error: ${{ matrix.os == 'macOS-latest' }}
     strategy:
+      fail-fast: false
```
Comment (ddn0) [inline .github/workflows/cpp.yaml:75]: I will miss the fail-fast option. I suppose more reason to fix the root cause.

Comment (arthurp) [reply .github/workflows/cpp.yaml:75]: Me too. Roll it back whenever you feel like it of course. It's just better to be able to merge PRs without forcing things, for the moment.

Comment (arthurp) [reply .github/workflows/cpp.yaml:75]: I wonder if we should pull out the MacOS builds into a separate work flow. That way we can have fast fail, but still let MacOS fail and still be able to push.  Maybe a project for another day.

Comment (ddn0) [reply .github/workflows/cpp.yaml:75]: Yeah. I think I'd rather just fix the MacOS build at that point.

Comment (arthurp) [reply .github/workflows/cpp.yaml:75]: I guess what I'm thinking is that having the MacOS builds marked as nor required doesn't mean much if when they fail they also cause all the other builds to be cancelled. ;-)

Comment (ddn0) [reply .github/workflows/cpp.yaml:75]: oh yeah. It was more an oversight on my part. I just didn't get around to making macos builds required.

**Pull request merged**